### ISSUE TITLE
Create accessibility component for hiding text visually

### ISSUE THREAD

**Issue by lymancheung** [telus/tds-core#532]
## Description

* With the introduction of TDS 2.0.0, the deprecated class `.accessibility-hide` has no current replacement
* There is currently no standardized class or component to visually hide text while making context available to assistive technology

## Recommendation

* Introduce a component `<A11yHide>` to visually hide text while still making context available

## Meta

* TDS version: v2.0.0
* Willing to develop solution: Yes (once time frees up)
* Has workaround: Yes (temporarily added the helper class into custom stylesheet)
* High impact: No

**Comment by ryanoglesby08** [comment]
Thanks @lymancheung! I think this sounds like a good idea 💡. I'm going to bring it to the TDS team and get their thoughts first, but I think this could be added to tds-core.

If we do proceed with this one, I'd like to sit with you to observe/help with the contribution process. We have been trying to improve our tooling a lot lately, and haven't added a net new component in a little while. So this could be a great chance to work through some kinks.

It also shouldn't be too much of a time commitment, maybe just a couple of hours to get the PR made. 

When do you think you may be able to find some free time?

**Comment by ryanoglesby08** [comment]
To address Oskar's comment, we would need to change/expand the interface for CTAs to allow for hidden text.

```jsx
<Button variant="primary" a11yText="Buy a new iPhone 10 now.">Buy now</Button>
// or
<Button variant="primary">
  Buy now
  <A11yContent>Buy a new iPhone 10 now</A11yContent>
</Button>
```

We will need to consider the best interface here that follows TDS conventions and principles.

**Comment by jraff** [comment]
This has been released in [@tds/core-a11y-content@1.0.0](https://github.com/telus/tds-core/releases/tag/%40tds%2Fcore-a11y-content%401.0.0)
